event_sidecar/src/admin_server.rs

```
@@ -37,14 +39,20 @@ impl AdminServer {
     }
 }
 
-pub async fn run_server(config: AdminServerConfig) -> Result<(), Error> {
-    AdminServer {
-        port: config.port,
-        max_concurrent_requests: config.max_concurrent_requests,
-        max_requests_per_second: config.max_requests_per_second,
+pub async fn run_server(config: AdminApiServerConfig) -> Result<ExitCode, Error> {
+    if config.enable_server {
+        AdminServer {
+            port: config.port,
+            max_concurrent_requests: config.max_concurrent_requests,
+            max_requests_per_second: config.max_requests_per_second,
+        }
+        .start()
+        .await
+        .map(|_| ExitCode::SUCCESS)
```

We could add a debug log with output here to avoid masking errors.

---

start returns Result<(), Error>. `map` handles the left part, I don't think there is any error masking here